mdd/python/translation/openconfig_xe/xe_qos.py

```diff
@@ -0,0 +1,281 @@
+# -*- mode: python; python-indent: 4 -*-
+from translation.openconfig_xe.common import xe_system_get_interface_ip_address
+from translation.common import get_interface_type_and_number
+
+dscp_dict = {8:'cs1', 10:'af11', 12:'af12', 14:'af13', 16:'cs2', 18:'af21', 20:'af22',
+                22:'af23', 24:'cs3', 26:'af31', 28:'af32', 30:'af33', 32:'cs4', 34:'af41',
+                36:'af42', 38:'af43', 40:'cs5', 46:'ef', 48:'cs6', 56:'cs7', 0:'default'}
+
+def xe_qos_program_service(self, nso_props) -> None:
+    """
+    Program service
+    """
+
+    device_cdb = nso_props.root.devices.device[nso_props.device_name].config
+
+    # Forwarding-groups
+    """
+    Configure forwarding-groups
+    """
+    for fg in nso_props.service.oc_qos__qos.forwarding_groups.forwarding_group:
+        device_cdb.ios__policy_map.create(fg.name)
+    if len(nso_props.service.oc_qos__qos.forwarding_groups.forwarding_group) == 0:
+        if device_cdb.ios__policy_map:
```

Comment:
Minor, but you can just combine these two if statements since you're checking both for truthy.
```suggestion
       if (len(nso_props.service.oc_qos__qos.forwarding_groups.forwarding_group) == 0 
           and device_cdb.ios__policy_map):
```